src/main/java/org/cbioportal/persistence/helper/StudyViewFilterHelper.java

```diff
@@ -138,29 +143,29 @@ public static <T extends DataFilter> List<T> mergeDataFilters(List<T> filters) {
                 // leave non-numerical values as they are
                 if (dataFilterValue.getValue() != null) {
                     nonNumericalValues.add(dataFilterValue);
+                    continue;
```

Comment:
@fuzhaoyuan can we add a comment explaining why non-null means non numerical?  i know that is true i just think someone coming to this for the first time would be baffled by it.

Reply:
also, a comment explaining why we're going to jump to next iteration (continue)

Reply:
i actually think that making an else clause would be clearer/preferable to continue.

Reply:
in general i think this logic here is very difficult to grok.   because the number of items (bins) is always going to be negligible, i would prioritize readability and adopt a functional approach where we just filter out the non numericals and numericals into a separate array, sort the numericals and then use reduce to iteratively merge them depending on adjacency.

Reply:
I used continue not else because otherwise SonarQube would complain the cognitive complexity.
I have added comments throughout the process. Basically yes we filter out non-numericals and then for numericals we merge them depending on adjacency.
The sorting would be a hassle because of the structure of class DataFilterValue, the actual bin range is hidden deep inside, so I just prevented unsorted filters in the input validation.